### PR TITLE
fix #764 - support external links in sidebar

### LINKED ISSUE #764: External urls in the sidebar
#### What problem does this feature solve?
Allows users to put external URLs in the sidebar
#### What does the proposed API look like?
Not sure
#### How should this be implemented in your opinion?
I would do it exactly the same as header menu items `{text: 'foo', link: 'google.com'}`
#### Are you willing to work on this yourself?**
Yes, point me in the right direction :)

### PR DESCRIPTION
**Pull request by timaschew**
Credits goes to #1012 
This PR is based on top of master and has no changes in the **Sidebar.vue** file compared to the original.

**Summary**

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:
**before** sidebar is not rendered at all, because of an error

**after** sidebar is rendered with external links properly

See demo: https://hungry-yonath-bfd0a1.netlify.com/

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [ ] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: packages/@vuepress/theme-default/components/SidebarLink.vue [hunk 1]
```diff
@@ -75,6 +77,18 @@ function renderChildren (h, children, path, route, maxDepth, depth = 1) {
     ])
   }))
 }
+
+function renderExternal (h, to, text) {
+  return h('a', {
+    attrs: {
+      href: to,
+      target: '_blank'
```
Comment (ulivz) [inline packages/@vuepress/theme-default/components/SidebarLink.vue]: Added `rel: 'noopener noreferrer'` here.

Comment (timaschew) [reply packages/@vuepress/theme-default/components/SidebarLink.vue]: done

File: packages/@vuepress/theme-default/util/index.js [hunk 2]
```diff
@@ -54,6 +54,13 @@ export function isActive (route, path) {
 }
 
 export function resolvePage (pages, rawPath, base) {
+  if (isExternal(rawPath)) {
+    return {
+      type: 'external',
+      title: rawPath,
```
Comment (ulivz) [inline packages/@vuepress/theme-default/util/index.js]: It looks strange to assign `rawPath` to `title`, you can omit it.

Comment (timaschew) [reply packages/@vuepress/theme-default/util/index.js]: Good point. Done

**Comment by narration-sd** [comment]
This is great, @timaschew & @ulivz, kind thanks for pushing this ahead  -- and I'd like to put it right away to good use.

I'm trying variations on this:
```
    npm install vuejs/vuepress#141bd11 --save-dev
```
...and in fact that incantation seems to find your work, as it gives back the full commit id, but then fails as:
```
npm ERR! Can't install github:vuejs/vuepress#141bd11e4b7e1b4d1bee53a49c232b11a36516cd: Missing package name
```
Digging on that Missing package name error, I ended up [here](https://npm.community/t/missing-package-name-error-when-installing-from-github/4471), which is not promising on its own.

How would you propose I pick this up, other than editing it into my node_modules or global source??

**Comment by rasapetter** [comment]
Author of the original PR here. Thank you for adopting my branch! Great to have this merged.

I started rebasing the branch a couple of days ago when I had a few minutes to spare, but the conflicts looked gnarly enough that I didn't follow through. Glad to see it wasn't as bad as it seemed.

**Pull request merged**